docs/irdl.ipynb

```diff
@@ -22,6 +22,46 @@
   {
    "cell_type": "code",
    "execution_count": 1,
+   "id": "1c670f99",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Requirement already satisfied: pytest<8.0 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from -r ../requirements.txt (line 1)) (7.2.0)\n",
+      "Requirement already satisfied: filecheck<0.0.23 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from -r ../requirements.txt (line 2)) (0.0.22)\n",
+      "Requirement already satisfied: lit<16.0.0 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from -r ../requirements.txt (line 3)) (15.0.6)\n",
+      "Requirement already satisfied: frozenlist<1.4.* in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from -r ../requirements.txt (line 4)) (1.3.3)\n",
+      "Requirement already satisfied: coverage<8.0.0 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from -r ../requirements.txt (line 5)) (7.0.4)\n",
+      "Requirement already satisfied: attrs>=19.2.0 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from pytest<8.0->-r ../requirements.txt (line 1)) (22.2.0)\n",
+      "Requirement already satisfied: iniconfig in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from pytest<8.0->-r ../requirements.txt (line 1)) (2.0.0)\n",
+      "Requirement already satisfied: packaging in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from pytest<8.0->-r ../requirements.txt (line 1)) (23.0)\n",
+      "Requirement already satisfied: pluggy<2.0,>=0.12 in /Users/sasha/Develop/xdslproject/xdsl/venv/lib/python3.11/site-packages (from pytest<8.0->-r ../requirements.txt (line 1)) (1.0.0)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Install requirements if necessary\n",
+    "%pip install -r ../requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "id": "7ac54610",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add the local xdsl definition to the path\n",
```

Comment:
So if I understand well, this works both in Pyodide and a local Jupyter?

Reply:
Indeed!
It followed from a discussion with @superlopuh who likes to use the notebooks to check things incrementally during development.
Using that allows to import directly from the sources, so the local code changes take effect.
Maybe an installed xDSL would shadow it though, I'm not sure about that.